mysticeti-core/src/syncer.rs

```diff
@@ -50,16 +72,24 @@ impl<H: BlockHandler, S: SyncerSignals, C: CommitObserver> Syncer<H, S, C> {
             .metrics
             .utilization_timer
             .utilization_timer("Syncer::add_blocks");
+        let previous_round = self.core().current_round();
         self.core.add_blocks(blocks);
-        self.try_new_block(connected_authorities.clone());
+        let new_round = self.core().current_round();
+
+        // we got a new quorum of blocks. Let leader timeout task know about it.
+        if new_round > previous_round {
+            self.round_advanced_signal.new_round(new_round);
+        }
+
+        self.try_new_block(connected_authorities);
     }
 
     pub fn force_new_block(
         &mut self,
         round: RoundNumber,
         connected_authorities: AuthoritySet,
     ) -> bool {
-        if self.core.last_proposed() == round {
+        if self.core.last_proposed() < round {
```

Comment:
why is this `<` and not `<=` I thought it was okay to propose multiple blocks per round